### PR TITLE
[Tooling] Use shared pipeline vars

### PR DESCRIPTION
**Pull request by iangmaia**
This PR updates the pipelines to use common values defined in the file `.buildkite/shared-pipeline-vars` (see paaHJt-5Rr-p2).

**Note**: This PR will fail until we update the setup also on the Buildkite side.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -665,7 +665,7 @@ platform :ios do
     zip(path: lane_context[SharedValues::XCODEBUILD_ARCHIVE], output_path: archive_zip_path)
 
     version = options[:beta_release] ? build_code_current : release_version_current
-    create_release(
+    create_github_release(
```
Comment (iangmaia) [inline fastlane/Fastfile:668]: ⚠️ Updating the call to the renamed `release-toolkit` action.

The PR https://github.com/woocommerce/woocommerce-ios/pull/13791 bumped the `release-toolkit` version to `12.0.0` and this was one of the breaking changes.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
-    - automattic/a8c-ci-toolkit#3.3.0
-  # Common environment values to use with the `env` key.
-  - &common_env
-    # Be sure to also update the `.xcode-version` file when updating the Xcode image/version here
-    IMAGE_ID: xcode-15.4
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+---
+
+agents:
+  queue: "mac"
```
Comment (spencertransier) [inline .buildkite/pipeline.yml:5]: For this and the other instances of `"mac"`: I noticed in other repos that we haven't added quotation marks around `mac` ([PCiOS](https://github.com/Automattic/pocket-casts-ios/blob/trunk/.buildkite/pipeline.yml), [WPiOS](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/.buildkite/pipeline.yml)). I assume that adding them here wouldn't cause any issues and is probably a better practice than without quotes, but wanted to bring it up just in case.

File: .buildkite/release-pipelines/publish-release.yml [hunk 3]
```diff
@@ -1,21 +1,26 @@
...
 
-      echo '--- :ruby: Setup Ruby Tools'
-      install_gems
+      echo '--- :closed_lock_with_key: Access Secrets'
+      bundle exec fastlane run configure_apply
 
       echo '--- :package: Publish Release'
       bundle exec fastlane publish_release skip_confirm:true
-    agents:
-      queue: "tumblr-metal"
```
Comment (spencertransier) [inline .buildkite/release-pipelines/publish-release.yml:18]: Why the change from `tumblr-metal` to the `mac` agents?

File: .buildkite/release-pipelines/publish-release.yml [hunk 4]
```diff
@@ -1,21 +1,26 @@
...
+  IMAGE_ID: $IMAGE_ID
+
 steps:
   - label: "Publish Release"
-    plugins:
-      - $CI_TOOLKIT
+    plugins: [$CI_TOOLKIT]
     command: |
-      echo '--- :robot_face: Use bot for git operations'
-      source use-bot-for-git
```
Comment (spencertransier) [inline .buildkite/release-pipelines/publish-release.yml:7]: I think I missed where we started using `use-bot-for-git`. Do the commands in `configure-git-for-release-management.sh` replace the need for that?

**Review by spencertransier — APPROVED** [review]
Looks great, just had a few questions 👍

File: .buildkite/release-pipelines/publish-release.yml (hunk 3 above)
Comment (iangmaia) [reply .buildkite/release-pipelines/publish-release.yml:18]: The point of using `tumblr-metal` for release tasks is to allow for them to run on a trusted agent that has the proper credentials (via using `use-bot-for-git`).
Unfortunately, the solution doesn't work for some macOS release tasks that need macOS tooling (e.g. to generate strings).

On this specific case, it actually wouldn't make a difference because the lane `publish_release` doesn't run anything that requires a Mac, so I guess it would just work on `tumblr-metal` (~but without using `use-bot-for-git`, as it only supports a pre-defined set of repos.~ See paaHJt-6EP-p2).

To make it consistent with the other release pipelines I decided to let it run on the same `mac` queue for now, until we build a Mac solution for trusted agents and all release tasks can run without needing a repo deploy key.

File: .buildkite/release-pipelines/publish-release.yml (hunk 4 above)
Comment (iangmaia) [reply .buildkite/release-pipelines/publish-release.yml:7]: Yeah, I already mentioned in my previous comment, but for now `use-bot-for-git` only works on Android repos. Se paaHJt-6EP-p2.
The other commands just replicate whatever the other release pipelines are doing.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (iangmaia) [reply .buildkite/pipeline.yml:5]: True... I think inlined strings in Yaml files don't really require the quotes. I think we're not really consistent there and I'm personally indifferent, but I think I got the pattern for WCiOS from WCAndroid 🤷

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/222">20.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (mokagio) [reply .buildkite/pipeline.yml:5]: I personally am a radical when it comes to `""` in YAML.

If they are not required (e.g. when a value includes `:` or starts with `*`) I don't use them. I feel it makes the YAML look leaner and easier to read.

But ☝️ I wouldn't enforce it as a rule unless we had a formatter that can apply it for us.

File: .buildkite/release-pipelines/publish-release.yml (hunk 3 above)
Comment (AliSoftware) [reply .buildkite/release-pipelines/publish-release.yml:18]: @iangmaia I'm not sure I follow.

As long as the `wpmobilebot` (whose SSH key is already installed in the trusted tumblr-metal agent) also has access to the `woocommerce-ios` repo (just like it already has access to `woocommerce-android`, we should be able to use `tumblr-metal` in this pipeline and call `use-bot-for-git wpmobilebot` to tell it to use that bot's key explicitly and this should work, no?

Sure, `woocommerce-ios` is [not one of the repos for which our `use-bot-for-git` helper would auto-detect the name of the bot user and SSH key to use](https://github.com/Automattic/buildkite-ci/blob/trunk/src/agents/tumblr-metal/resources/use-bot-for-git.sh#L40), but the whole reason why [we added the possibility to pass an extra parameter to this script](https://github.com/Automattic/buildkite-ci/blob/trunk/src/agents/tumblr-metal/resources/use-bot-for-git.sh#L35-L37) was to be able to support other repos that were not yet in the auto-detect list… without having to redeploy the `tumblr-metal` agent every time we wanted to use it for a new repo, right?

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (AliSoftware) [reply .buildkite/pipeline.yml:5]: Same as Gio, I feel like the YAML is more readable without the extra quotes if they are not necessary for YAML syntax disambiguation.

Besides, not wrapping the string in quotes means we can add quotes within the string if needed without having to escape them (e.g. `command: .buildkite/commands/run-my-script.sh "Hello world"` vs `command: ".buildkite/commands/run-my-script.sh \"Hello world\""`)

But same as Gio, while I like if we can be consistent across apps and files because it's always nicer to be consistent, I wouldn't be too strict about enforcing this particular rule either 🙃

File: .buildkite/release-pipelines/publish-release.yml (hunk 3 above)
Comment (iangmaia) [reply .buildkite/release-pipelines/publish-release.yml:18]: Had forgotten about the `use-bot-for-git` parameter option, so yes it would work as it doesn't have anything macOS specific as I mentioned.
At the same time, my main goal with the change was to keep it consistent with all the other release pipelines -- even when we eventually have a trusted agent solution for iOS, it would always be puzzling why this single one is running on a different agent.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (iangmaia) [reply .buildkite/pipeline.yml:5]: Agreed  🤝 @mokagio @AliSoftware @spencertransier. Moved from the _indifferent_ camp to the _if they're not required, don't use the quotes_ camp.

I'll send a small PR for WCiOS while at it and whenever I touch these files I'll remove unneeded quotes 👍

File: .buildkite/release-pipelines/publish-release.yml (hunk 3 above)
Comment (AliSoftware) [reply .buildkite/release-pipelines/publish-release.yml:18]: I mean I get it, but running steps that require git push access on `tumblr-metal` even on a iOS/Mac pipeline is already what we do on other repos (recent case in point: https://github.com/Automattic/simplenote-ios/pull/1650), so… 🤷
